fetch/manip.go

```diff
@@ -83,7 +84,20 @@ func Resize(src io.Reader, c *CacheContext) (io.Reader, error) {
 	}
 
 	if c.Crop {
-		options.Height = c.Width
+		data := bytes.NewReader(raw)
+
+		image, _, err := image.Decode(data)
+		if err != nil {
+			return nil, err
+		}
+
+		minDimension := int(math.Min(float64(image.Bounds().Size().X), float64(image.Bounds().Size().Y)))
+
+		if minDimension < options.Width || options.Width == 0 {
```

Comment:
It wasn't obvious (to me) these were the right return values, since I didn't actually know if sending `c=true` but no size returned any particular error or not. So I just tried it out in a playground to prove it to myself.